src/Microsoft.TemplateEngine.Abstractions/IVariableCollectionEx.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.TemplateEngine.Abstractions.Parameters;
+
+namespace Microsoft.TemplateEngine.Abstractions
+{
+    /// <summary>
+    /// Extends the <see cref="IVariableCollection"/> contract with the metadata
+    /// about data from template parameters.
+    /// </summary>
+    public interface IVariableCollectionEx : IVariableCollection
```

Comment:
Another thought: should it inherit from `IVariableCollection`? 
If we decide not to extend `IMacro` interface, probably it can stay isolated.

Reply:
It still needss to inherit so that we can pass it through without the change of contract (so that CoalesceMacro can cast it).
Other than that it can stay internal to RP project